src/Data/Algorand/Block.hs

```diff
@@ -0,0 +1,192 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+-- | Algorand block.
+module Data.Algorand.Block
+  ( Block (..)
+  , Rewards (..)
+  , UpgradeState (..)
+  , UpgradeVote (..)
+  , BlockWrapped (..)
+  , BlockHash
+  , Round (..)
+  ) where
+
+import qualified Data.Text as T
+
+import Data.Aeson (FromJSON, ToJSON)
+import Data.ByteArray (Bytes)
+import Data.ByteArray.Sized (SizedByteArray)
+import Data.Text (Text)
+import Data.Time (UTCTime)
+import Data.Time.Clock.POSIX (posixSecondsToUTCTime)
+import Data.Word (Word64)
+import GHC.Generics (Generic)
+import Servant.API (ToHttpApiData (toQueryParam))
+
+import Data.Algorand.Amount (Microalgos)
+import Data.Algorand.MessagePack (AlgoMessagePack (..), Canonical (..), MessageUnpackObject (..),
+                                  (.:), (.:>), (.:?))
+import Data.Algorand.Transaction (GenesisHash)
+import Data.Algorand.Transaction.Signed (SignedTransaction)
+
+type BlockHash = SizedByteArray 32 Bytes
+type Seed = SizedByteArray 32 Bytes
+type TransactionsRoot = SizedByteArray 32 Bytes
+type Addr = SizedByteArray 32 Bytes
+
+newtype Round = Round { unRound :: Word64 }
+  deriving stock (Eq, Ord, Show)
+  deriving newtype (ToJSON, FromJSON)
+
+instance AlgoMessagePack Round where
+  toAlgoObject = toAlgoObject . unRound
+  fromAlgoObject = fmap Round . fromAlgoObject
+
+instance ToHttpApiData Round where
+  toQueryParam (Round r) = T.pack $ show r
```

Comment:
I am not sure, but I guess it could be automatically derived using `newtype` strategy because `Word64` is most likely an instance of `ToHttpApiData` (I'll be surprised if it's not).

The same should apply to `AlgoMessagePack` instance, it looks as what would be produced by newtype deriving.

Reply:
It's working with `ToHttpData`, but not with `AlgoMessagePack`

<img width="1178" alt="image" src="https://user-images.githubusercontent.com/6579275/126766469-7f6b6119-b4b5-42f7-8121-efb5ba689246.png">

Reply:
Oh, okay, then I guess it's easier to just write `AlgoMessagePack` instance manually. I somewhat dislike the type of `fromAlgoObject`, but ok.